inc/class-wpseo-admin-bar-menu.php

```diff
@@ -464,17 +464,21 @@ protected function get_singular_post() {
 		return $post;
 	}
 
-	/**
-	 * Gets the focus keyword for a given post.
-	 *
-	 * @param WP_Post $post Post object to get its focus keyword.
-	 *
-	 * @return string Focus keyword, or empty string if none available.
-	 */
-	protected function get_post_focus_keyword( WP_Post $post ) {
-		if ( apply_filters( 'wpseo_use_page_analysis', true ) !== true ) {
-			return '';
-		}
+    /**
+     * Gets the focus keyword for a given post.
+     *
+     * @param $post Post object to get its focus keyword.
+     *
+     * @return string Focus keyword, or empty string if none available.
+     */
+    protected function get_post_focus_keyword( $post ) {
+        if ( ! is_object( $post ) || ! property_exists( $post, 'ID' ) ) {
```

Comment:
You might consider leaving the `is_object` check because this method is called in a situation where an `is_object` check already has been done. See `get_singular_post` which is doing the `is_object` check.

Reply:
Discussed and we'll keep it as it is.